### PR TITLE
Delete buttons

### PR DESCRIPTION
**Pull request by NunoDasNeves**
Needs prettifying and critique.

- meta field with canDelete, canEdit is now passed by the backend on requests for all kind of posts (comments, questions, reviews)
- due to this (and for future similar issues with refreshing the page etc) there is a thing in the router which will block on boot until checkAuth is completed. A slightly more refined version of this may be needed, but it might be okay for now (a better version would show the loading circle during checkAuth, and allow requests that really don't need auth)
- working delete buttons for all kinds of posts with confirm() cos deleting is a big deal. Button should be moved to triple dot menu thing and made pretty
- no edit functionality yet, but most of the infrastructure is there in the store, i think save it for a future PR
- thoughts on delete/edit buttons being in course feeds? At the moment it's just in Question and Review views.
- some bugfixes and random improvements included
- thoughts on mass delete/edit/something interface? moderation interface showing reported posts to review?

### CODE REVIEW

File: frontend/src/views/Question.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <AppBreadCrumb/>
 
     <transition name="fade-slide">
-      <QuestionCard v-bind="question" :authenticated="isLoggedIn" v-if="!loadingQuestion" />
+      <QuestionCard v-bind="question" :code="code" :authenticated="isLoggedIn" v-if="!loadingQuestion" />
```
Comment (DarkPurple141) [inline frontend/src/views/Question.vue]: ?

**Comment by DarkPurple141** [comment]
> * thoughts on delete/edit buttons being in course feeds? At the moment it's just in Question and Review views.
I think for now, not.
> * thoughts on mass delete/edit/something interface? moderation interface showing reported posts to review?
This is a good idea. Wondering if backend should employ sqlserver `triggers` for delete.

File: frontend/src/components/Questions/QuestionCard.vue [hunk 2]
```diff
@@ -23,9 +24,29 @@ export default {
     body: String,
     // TODO fix
     published: String,
-    authenticated: Boolean
+    authenticated: Boolean,
+    meta: Object
+  },
+  computed: {
+    menu() {
+      let options = []
```
[the remark this answers is not in the document]
Comment (NunoDasNeves) [reply frontend/src/components/Questions/QuestionCard.vue]: yup

File: frontend/src/store/auth/index.js [hunk 3]
```diff
@@ -184,6 +188,10 @@ const actions = {
       commit('ERROR', error.message)
     }
 
+    // signal the CV so the app can continue loading and use the JWT token in its requests
```
[the remark this answers is not in the document]
Comment (DarkPurple141) [reply frontend/src/store/auth/index.js]: Sorry as in, a vuex commit. The bit underneath that alters the state.

File: frontend/src/views/Question.vue (hunk 1 above)
Comment (DarkPurple141) [reply frontend/src/views/Question.vue]: Maybe. ^

File: frontend/src/store/auth/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (NunoDasNeves) [reply frontend/src/store/auth/index.js]: ooooh ya

**Pull request merged**